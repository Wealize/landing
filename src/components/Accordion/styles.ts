import styled from 'styled-components'

import { sixthPx, halfPx, twoThirdPx, singlePx, singleAndHalfPx, doublePx, doubleAndHalfPx, triplePx, quadruplePx, quadruple } from '../../theme/space'
import { media } from '../../theme/media'

export const Container = styled.div<{ backgroundColor: string, textColor: string}>`
  width: 100%;
  min-height: 20vh;
  padding: ${doubleAndHalfPx()} ${singlePx()};

  ${media.greaterThan('md')`
    padding: ${triplePx()} ${quadruplePx()};
  `};

  ${media.greaterThan('lg')`
    padding: ${quadruplePx()} ${quadruple() * 1.5}px;
  `};

  ${media.greaterThan('xl')`
    padding: ${quadruplePx()} ${quadruple() * 2.5}px;
  `};

  ${media.greaterThan('xxl')`
    padding: ${quadruplePx()} ${quadruple() * 3.5}px;
  `};

  color: ${({ textColor }) => textColor};
  background-color: ${({ backgroundColor }) => backgroundColor};
`

export const TitleContainer = styled.div`
  padding-bottom: 0;

  > .accordion-separator{
    margin-top: ${halfPx()};
  }
`

export const Title = styled.h5<{ textColor: string}>`
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: bold;
  color: ${({ textColor }) => textColor};

  ${media.greaterThan('xl')`
    font-size: 1.1rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 1.2rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 1.2rem;
  `};
`

export const Separator = styled.hr<{ textColor: string}>`
  width: 100%;
  background-color: ${({ textColor }) => textColor};
  opacity: .6;
  border: none;
  height: 2px;
  margin: 0;
`

export const SectionContainer = styled.div`
  padding-top: ${singleAndHalfPx()};

  > .accordion-separator {
    margin-top: ${singleAndHalfPx()};
  }

  ${media.greaterThan('md')`
    padding-top: ${doublePx()};

    > .accordion-separator {
      margin-top: ${doublePx()};
    }
  `};
`

export const SectionHeader = styled.div<{isExpanded: boolean}>`
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 0;

  > .button-rounded {
    display: none;

    & span {
      font-weight: lighter;
    }
  }


  ${media.greaterThan('lg')`
    > .accordion-section-toggle-button {
      display: none;
    }

    > .button-rounded {
      display: block;
      min-height: 75px;
    }
  `};

`

export const SectionTitle = styled.h6`
  font-size: 1.7rem;
  width: 75%;
  line-height: 1.52;

  ${media.greaterThan('md')`
    font-size: 1.8rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 2.2em;
  `};

  ${media.greaterThan('xl')`
    font-size: 2.4rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 2.7rem;
  `};
`

export const SectionToggle = styled.button<{
  textColor: string
  backgroundColor: string
  isExpanded: boolean
}>`
  width: 40px;
  height: 40px;
  background-color: transparent;
  border: none;
  padding: 0;
  transition: all 0.5s ease-out;
  transform: ${({ isExpanded }) =>
    isExpanded ? 'rotate(0deg)' : 'rotate(180deg)'};

  & svg > g:first-child {
    fill: ${({ textColor, isExpanded }) => isExpanded ? textColor : 'none'};
  }

  & svg > g > path:last-child {
    fill: ${({ textColor }) => textColor};
  }

  & svg > path {
    fill: ${({ textColor, backgroundColor, isExpanded }) => isExpanded ? backgroundColor : textColor};
  }

  ${media.greaterThan('md')`
      width: 50px;
      height: 50px;
  `};

  ${media.greaterThan('lg')`
    display: none;
  `};
`

export const SectionContentContainer = styled.div<{isExpanded: boolean}>`
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: ${singlePx()};
  transition: all 0.5s ease-out;
  overflow: hidden;
  height: auto;
  max-height: ${({ isExpanded }) => (isExpanded ? 1000 : 0)}px;
  padding-top: ${({ isExpanded }) => (isExpanded ? singleAndHalfPx() : 0)};

  ${media.greaterThan('lg')`
    grid-template-columns: repeat(3, 1fr);
  `};

  ${media.greaterThan('xl')`
    grid-template-columns: repeat(3, 1fr);
  `};

  ${media.greaterThan('xxl')`
    grid-template-columns: repeat(4, 1fr);
  `};

`

export const SectionElement = styled.span`
  padding: ${twoThirdPx()} 0 ${sixthPx()};
  line-height: 1.4;

  &:before {
    content: '- '
  }

  ${media.greaterThan('md')`
    font-size: 1.2rem;
    padding: ${twoThirdPx()};
  `};

  ${media.greaterThan('lg')`
    font-size: 1.3rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 1.4rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 1.5rem;
  `};
`
