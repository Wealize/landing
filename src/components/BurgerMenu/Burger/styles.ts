import styled from 'styled-components'

import { ACCENT_COLOR } from '../../../theme/color'
import { media } from '../../../theme/media'
import { singlePx } from '../../../theme/space'

export const StyledBurger = styled.button<{ open: boolean }>`
  top: ${singlePx()};
  margin-top: 2.5%;
  left: ${({ open }) => (open ? '85%' : '85%')};
  width: 2rem;
  height: 1.5rem;
  position: fixed;
  background: transparent;
  z-index: 10;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  border: none;
  cursor: pointer;
  padding: 0;
  &:focus {
    outline: none;
  }

  ${media.greaterThan('md')`
    margin-top: 0.4%;
  `}

  ${media.greaterThan('lg')`
    display: none;
  `}

  div {
    width: 1.5rem;
    height: 0.18rem;
    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;
    background-color: ${ACCENT_COLOR};

    :first-child {
      transform: ${({ open }) => (open ? 'rotate(45deg)' : 'rotate(0)')};
    }
    :nth-child(2) {
      opacity: ${({ open }) => (open ? '0' : '1')};
      transform: ${({ open }) => (open ? 'translateX(20px)' : 'translateX(0)')};
    }
    :nth-child(3) {
      transform: ${({ open }) => (open ? 'rotate(-45deg)' : 'rotate(0)')};
    }
  }
`
