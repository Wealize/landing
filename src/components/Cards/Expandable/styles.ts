import styled from 'styled-components'

import {
  singlePx,
  singleAndHalfPx,
  doublePx,
  triplePx,
  quadruplePx,
  quadruple
} from '../../../theme/space'
import { media } from '../../../theme/media'

export const Container = styled.div<{
  backgroundColor: string
  textColor: string
  isExpanded: boolean
}>`
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: ${singleAndHalfPx()} ${singlePx()};
  background-color: ${({ backgroundColor }) => backgroundColor};
  color: ${({ textColor }) => textColor};
  transition: all 0.5s ease-out;

  ${media.greaterThan('md')`
    padding: ${singleAndHalfPx()} ${quadruple() * 2}px;
  `};

  ${media.greaterThan('lg')`
    padding: ${quadruplePx()} ${doublePx()};
    min-height: 900px;
    min-width: 110%;
    justify-content: space-between;
  `};

  ${media.greaterThan('xl')`
    min-height: 980px;
  `};

  ${media.greaterThan('xxl')`
    padding: ${quadruplePx()};
    min-height: 1100px;
  `};
`

export const Header = styled.div<{
  isExpanded: boolean
}>`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-bottom: ${({ isExpanded }) => (isExpanded ? singlePx() : 0)}px;
`

export const Content = styled.div<{
  isExpanded: boolean
}>`
  flex-direction: column;
  justify-content: center;
  transition: max-height 0.5s ease-out;
  overflow: hidden;
  height: auto;
  max-height: ${({ isExpanded }) => (isExpanded ? 1000 : 0)}px;

  ${media.greaterThan('lg')`
    max-height: 1000px;
  `};
`

export const Headline = styled.h3`
  font-size: 1.2rem;
  font-weight: bold;

  ${media.greaterThan('md')`
    font-size: 1.2rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 1.3rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 1.5rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 2rem;
  `};
`

export const Toggle = styled.button<{
  textColor: string
  isExpanded: boolean
}>`
  width: 40px;
  height: 40px;
  background-color: transparent;
  border: none;
  padding: 0;
  transition: all 0.5s ease-out;
  transform: ${({ isExpanded }) =>
    isExpanded ? 'rotate(0deg)' : 'rotate(180deg)'};

  & svg > g > path:last-child {
    fill: ${({ textColor }) => textColor};
  }

  & svg > path {
    fill: ${({ textColor }) => textColor};
  }

  ${media.greaterThan('md')`
      width: 50px;
      height: 50px;
  `};

  ${media.greaterThan('lg')`
    display: none;
  `};
`

export const ImageContainer = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: ${doublePx()} 0;

  & svg {
    width: 40%;

    ${media.greaterThan('md')`
      width: 55%;
    `};

    ${media.greaterThan('lg')`
      width: 65%;
    `};
  }
  ${media.greaterThan('md')`
    padding: ${triplePx()} 0;
  `};

  ${media.greaterThan('lg')`
    padding: ${quadruplePx()} 0;
  `};
`

export const Subheadline = styled.h4`
  display: flex;
  line-height: 1.2;
  flex-direction: row;
  justify-content: flex-start;
  font-size: 1.8rem;
  text-align: left;
  padding-bottom: ${singlePx()};

  ${media.greaterThan('md')`
    font-size: 1.7rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 1.8rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 2.2rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 2.5rem;
  `};
`

export const Description = styled.h5`
  font-size: 1.1rem;
  text-align: left;
  line-height: 1.43;

  ${media.greaterThan('md')`
    font-size: 1.3rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 1.3rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 1.7rem;
  `};
`
