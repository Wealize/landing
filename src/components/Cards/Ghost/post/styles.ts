import styled from 'styled-components'

import { WHITE_COLOR } from '../../../../theme/color'
import { singlePx } from '../../../../theme/space'

export const Container = styled.a`
  width: 100%;
  background: ${WHITE_COLOR};
  z-index: 1;
`

export const CardHeadline = styled.h4`
  font-size: 1rem;
  font-weight: bold;
  padding-bottom: ${singlePx()};
`

export const CardPublishedAt = styled.span`
  font-size: 1rem;
  padding: ${singlePx()} 0;
`
