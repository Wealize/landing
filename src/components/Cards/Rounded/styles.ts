import styled from 'styled-components'

import { PRIMARY, WHITE, DARK_GRAY, LIGHT_GRAY } from '../../../theme/color'
import {
  thirdPx,
  singlePx,
  singleAndHalfPx,
  doubleAndHalfPx
} from '../../../theme/space'
import { media } from '../../../theme/media'
import { DEFAULT_RADIUS } from '../../../theme/border'

export const Container = styled.article<{ hasPrimary: boolean }>`
  width: 100%;
  background: ${LIGHT_GRAY};
  display: flex;
  flex-direction: column;

  ${({ hasPrimary }) => media.greaterThan('md')`
    display: flex;
    flex-direction: row;
    flex-direction: ${hasPrimary ? 'row' : 'row-reverse'};
  `};
`

export const ImageContainer = styled.div<{ imageUrl: string }>`
  width: 100%;
  display: flex;
  align-items: stretch;
  background: url(${({ imageUrl }) => imageUrl}) no-repeat top center;
  background-size: cover;
  background-position: center;
  min-height: 290px;

  ${media.greaterThan('md')`
    width: 51%;
    min-height: 350px;
  `};

  ${media.greaterThan('lg')`
    min-height: 500px;
  `};
`

export const ContentContainer = styled.div<{ hasPrimary: boolean }>`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: -8%;
  top: 90%;
  background-color: ${({ hasPrimary }) => (hasPrimary ? PRIMARY : WHITE)};
  color: ${({ hasPrimary }) => (hasPrimary ? WHITE : DARK_GRAY)};
  padding: ${singlePx()};
  border-top-left-radius: ${DEFAULT_RADIUS}px;
  border-top-right-radius: ${DEFAULT_RADIUS}px;
  min-height: 350px;

  ${({ hasPrimary }) => media.greaterThan('md')`
    width: 45%;
    flex-direction: column-reverse;
    justify-content: space-between;
    margin-top: 0;
    display: flex;
    padding: ${singleAndHalfPx()};
    border-top-left-radius: ${hasPrimary ? 'initial' : `${DEFAULT_RADIUS}px`};
    border-top-right-radius: ${hasPrimary ? `${DEFAULT_RADIUS}px` : `initial`};
    min-height: 450px;
  `};

  ${media.greaterThan('lg')`
    padding: ${doubleAndHalfPx()};
    min-height: 550px;
  `};
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
`

export const Title = styled.h5`
  text-transform: uppercase;
  font-size: 1rem;
  font-weight: bold;
  padding: 0 0 ${singlePx()} 0;

  ${media.greaterThan('lg')`
    font-size: 1.5rem;
  `};
`

export const Description = styled.p`
  line-height: 1.5;
  font-size: 1rem;

  ${media.greaterThan('lg')`
    font-size: 1.2rem;
  `};
`

export const LinkContainer = styled.div<{ color: string }>`
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: ${singlePx()} 0;
  color: ${({ color }) => color};
`

export const LinkAnchor = styled.a<{ color: string }>`
  display: flex;
  flex-direction: row;
  text-decoration: none;
  font-weight: bold;
  font-size: 1.2rem;
  margin-right: ${thirdPx()};
  color: ${({ color }) => color};
`

export const TagsContainer = styled.div`
  padding-top: ${singlePx()};

  ${media.greaterThan('md')`
    order: 1;
    padding: 0 0 ${singlePx()} 0;
  `};
`

export const Tag = styled.span<{ hasPrimary: boolean }>`
  display: inline-block;
  text-align: center;
  padding: ${thirdPx()};
  border-radius: 16px;
  border: 1px solid ${({ hasPrimary }) => (hasPrimary ? WHITE : DARK_GRAY)};
`
