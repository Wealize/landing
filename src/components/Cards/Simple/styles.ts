import styled from 'styled-components'

import { WHITE_COLOR, DARK_GRAY_COLOR } from '../../../theme/color'
import { media } from '../../../theme/media'
import { CONTAINER_SPACING } from '../../../theme/spacingGuide'

export const Container = styled.div`
  width: 100%;
  background: ${WHITE_COLOR};
  z-index: 1;
  ${CONTAINER_SPACING.MOBILE}

  ${media.greaterThan('md')`
    ${CONTAINER_SPACING.TABLET}
  `};

  ${media.greaterThan('lg')`
    ${CONTAINER_SPACING.SMALL_DESKTOP};
  `};

  ${media.greaterThan('xl')`
    ${CONTAINER_SPACING.DESKTOP};
  `};

  ${media.greaterThan('xxl')`
    ${CONTAINER_SPACING.WIDE};
  `};
`

export const Text = styled.h3`
  color: ${DARK_GRAY_COLOR};
  font-size: 2rem;
  line-height: 45px;
  word-wrap: break-word;

  ${media.greaterThan('md')`
    font-size: 3rem;
    line-height: 60px;
  `};

  ${media.greaterThan('lg')`
    max-width: 65%;
  `};

  ${media.greaterThan('xl')`
    max-width: 70%;
    line-height: 75px;
  `};

  ${media.greaterThan('xxl')`
    font-size: 3.5rem;
    max-width: 85%;
  `};
`
