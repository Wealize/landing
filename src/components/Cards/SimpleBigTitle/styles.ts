import styled from 'styled-components'

import {
  singlePx,
  doublePx
} from '../../../theme/space'
import { media } from '../../../theme/media'
import { CONTAINER_SPACING } from '../../../theme/spacingGuide'

export const Container = styled.div`
  ${CONTAINER_SPACING.MOBILE}

  ${media.greaterThan('md')`
    ${CONTAINER_SPACING.TABLET}
  `};

  ${media.greaterThan('lg')`
    ${CONTAINER_SPACING.SMALL_DESKTOP};
  `};

  ${media.greaterThan('xl')`
    ${CONTAINER_SPACING.DESKTOP};
  `};

  ${media.greaterThan('xxl')`
    ${CONTAINER_SPACING.WIDE};
  `};
`

export const Headline = styled.span`
  font-size: 3rem;
  line-height: 1.09;
  text-align: left;

  ${media.greaterThan('md')`
    padding: ${doublePx()} 0;
  `};

  ${media.greaterThan('lg')`
    font-size: 3.5rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 3.7rem;
  `};
`

export const SubHeadline = styled.h5`
  padding-top: ${singlePx()};
  line-height: 1.46;
  text-align: left;
  font-size: 1.7rem;

  ${media.greaterThan('md')`
    padding: ${doublePx()} 0;
    font-size: 1.8rem;
  `};
`
