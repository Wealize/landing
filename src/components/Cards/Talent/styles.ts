import styled from 'styled-components'

import { WHITE_COLOR, ACCENT_COLOR, GRAY_300_COLOR } from '../../../theme/color'
import { thirdPx, singlePx, doublePx, triplePx } from '../../../theme/space'
import { media } from '../../../theme/media'

export const Container = styled.div`
  width: 100%;
  background: ${WHITE_COLOR};
  padding: ${singlePx()};
  z-index: 1;
  min-width: 300px;
  min-height: 300px;

  ${media.greaterThan('md')`
    width: 100%;
    padding: ${doublePx()};
    min-width: 300px;
    min-height: 300px;
  `};

  ${media.greaterThan('lg')`
    padding: ${doublePx()};
    min-width: 200px;
    min-height: 200px;
  `};

  ${media.greaterThan('xl')`
    padding: ${triplePx()};
    min-width: 300px;
    min-height: 300px;
  `};
`

export const ImageContainer = styled.div`
  width: 100%;
  height: 100%;
  min-width: 300px;
  min-height: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  margin-bottom: ${singlePx()};

  & img {
    object-fit: cover;
  }

  ${media.greaterThan('md')`
    min-width: 300px;
    min-height: 300px;
  `};

  ${media.greaterThan('lg')`
    min-width: 250px;
    min-height: 250px;
  `};

  ${media.greaterThan('xl')`
    min-width: 350px;
    min-height: 350px;
  `};
`

export const ContentContainer = styled.div``

export const Headline = styled.h4`
  font-size: 1rem;
  text-align: left;
  font-weight: bold;
  padding-bottom: ${thirdPx()};

  ${media.greaterThan('md')`
    padding-bottom: ${thirdPx()};
    font-size: 1.2rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 1.2rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 1.5rem;
  `};
`

export const Subheadline = styled.h5`
  font-size: 1rem;
  text-align: left;
  font-weight: lighter;
  padding-bottom: ${thirdPx()};
  color: ${GRAY_300_COLOR};

  ${media.greaterThan('md')`
    font-size: 1.2rem;
    padding-bottom: ${thirdPx()};
  `};

  ${media.greaterThan('xl')`
    font-size: 1.5rem;
  `};
`

export const SocialNetworksContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;

  ${media.greaterThan('lg')`
    padding-top: ${thirdPx()};
  `};

  & > a {
    text-decoration: none;
    color: ${ACCENT_COLOR};
    font-weight: bold;
    text-transform: capitalize;
    margin-right: ${thirdPx()};

    ${media.greaterThan('md')`
    font-size: 1.2rem;
    `};

    ${media.greaterThan('xl')`
    font-size: 1.5rem;
  `};
  }
`
