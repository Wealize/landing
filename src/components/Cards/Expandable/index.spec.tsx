import React from 'react'
import { shallow } from 'enzyme'

import CardExpandable from './index'
import { Toggle, Content, Headline, Subheadline, Description } from './styles'
import BlockchainIcon from '../../icons/thingsWeAreGoodAt/BlockchainIcon'
import { PRIMARY, PRIMARY_LIGHT } from '../../../theme/color'

describe('<CardExpanded />', () => {
  const cardProps = {
    headline: 'headline',
    subheadline: 'subheadline',
    description: 'description',
    image: <BlockchainIcon />,
    backgroundColor: PRIMARY_LIGHT,
    textColor: PRIMARY,
    isExpanded: false
  }
  const wrapper = shallow(<CardExpandable {...cardProps} />)

  it('wrapper: should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render content correctly', () => {
    expect(wrapper.find(Toggle).length).toEqual(1)
    expect(wrapper.find(Content)).toHaveStyleRule('max-height', '0px')
    expect(wrapper.find(BlockchainIcon).length).toEqual(1)
    expect(wrapper.find(Headline).text()).toEqual(cardProps.headline)
    expect(wrapper.find(Subheadline).text()).toEqual(cardProps.subheadline)
    expect(wrapper.find(Description).text()).toEqual(cardProps.description)

    const cardPropsExpanded = {
      ...cardProps,
      isExpanded: true
    }
    const wrapperExpanded = shallow(<CardExpandable {...cardPropsExpanded} />)

    expect(wrapperExpanded.find(Toggle).length).toEqual(1)
    expect(wrapperExpanded.find(Content)).toHaveStyleRule(
      'max-height',
      '1000px'
    )
    expect(wrapperExpanded.find(BlockchainIcon).length).toEqual(1)
    expect(wrapperExpanded.find(Headline).text()).toEqual(cardProps.headline)
    expect(wrapperExpanded.find(Subheadline).text()).toEqual(
      cardProps.subheadline
    )
    expect(wrapperExpanded.find(Description).text()).toEqual(
      cardProps.description
    )
  })

  it('should click on toggle button', async () => {
    const mockOnClick = jest.fn()
    const toggleComponent = shallow(
      <Toggle onClick={mockOnClick} textColor={PRIMARY} isExpanded={false} />
    )
    toggleComponent.simulate('click')
    expect(mockOnClick.mock.calls.length).toEqual(1)
  })
})
