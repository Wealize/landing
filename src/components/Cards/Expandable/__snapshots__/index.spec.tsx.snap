// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardExpanded /> wrapper: should render 1`] = `
<styled.div
  backgroundColor="#ffc9c9"
  isExpanded={false}
  textColor="#f06078"
>
  <styled.div
    isExpanded={false}
  >
    <styled.h3>
      headline
    </styled.h3>
    <styled.button
      aria-label="expand card content button"
      isExpanded={false}
      onClick={[Function]}
      textColor="#f06078"
    >
      <Memo(RoundArrowIcon) />
    </styled.button>
  </styled.div>
  <styled.div
    isExpanded={false}
  >
    <styled.div>
      <Memo(BlockchainIcon) />
    </styled.div>
    <styled.h4>
      subheadline
    </styled.h4>
    <styled.h5>
      description
    </styled.h5>
  </styled.div>
</styled.div>
`;
